src/components/CalendarGrid.tsx

```diff
@@ -25,8 +25,8 @@ export function CalendarGrid({
   style,
   length = '2 hours',
 }: CalendarGridProps) {
-  const { view, viewTimes } = useCalendar();
-  const numDaysDisplayed = view === 'day' ? 1 : 7;
+  const { view, viewTimes, customDays } = useCalendar();
+  const numDaysDisplayed = view === 'day' ? 1 : customDays;
```

Comment:
There is a bug here: If the view is a week, the Grid only spans for `customDays` (you can see that in the example). This line should work:
```ts
const numDaysDisplayed = { week: 7, day: 1, custom: customDays }[view];
```
I am also wondering if it would help to introduce a property on the `CalendarContextType` that contains the number of days displayed, as it is used in many places and could simplify the code here and there. But this doesn't have to be done in this PR.

Reply:
I frankly don't understand the line you wrote. I fixed it but I really don't know what it is doing